src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -717,7 +717,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     current.OriginalDefinition.AddUseSiteDiagnostics(useSiteDiagnostics)
                 End If
 
-                If current = superType Then
+                If current.IsSameType(superType, TypeCompareKind.IgnoreTupleNames) Then
```

Comment:
I aligned the implementations and added a test. 
Just to confirm, we want a type to derive from another even if the modifiers are different?